### ISSUE TITLE
[IMPROVEMENT] instance-managers for v1 and v2 volumes respectively  

### ISSUE THREAD

**Issue by derekbit** [longhorn/longhorn#6984]
## Is your improvement request related to a feature? Please describe (👍 if you like this request)

To reduce the cross interference between v1 and v2 volumes, handling v1 and v2 volumes in separate pods is better. 
The ticket aims to create instance-managers for v1 and v2 volumes respectively. 

## Additional context

cc @shuo-wu @DamiaSan @innobead

**Comment by derekbit** [comment]
Waiting. for the merge of https://github.com/longhorn/longhorn-manager/pull/2371

**Comment by yangchiu** [comment]
Tested on master-head, same as https://github.com/longhorn/longhorn/issues/7324#issuecomment-1869865367. I can find new instance manager being created:
```
$ kubectl -n longhorn-system get lhim
NAME                                                STATE     TYPE   DATA ENGINE   NODE            AGE
instance-manager-3e15168c64517b1cb98628231c6ea0de   running   aio    v1            ip-10-0-1-104   3m5s
instance-manager-f9991e314952df13e283d4ffaa3dacaa   running   aio    v1            ip-10-0-1-111   3m5s
instance-manager-177e71eb8d1be212344f6b9711bf95bf   running   aio    v1            ip-10-0-1-151   3m5s
instance-manager-bc52bfb004f9895afa49b8cf19fcae66   running   aio    v2            ip-10-0-1-104   65s
instance-manager-df2fcbea2f4e3c6c307dcdac0be0e70a   running   aio    v2            ip-10-0-1-111   67s
instance-manager-7e9b887fa5c4bc918ac329ac4fc4ca6a   running   aio    v2            ip-10-0-1-151   67s
```
But v2 volume can't be created:
```
$ kubectl describe pvc longhorn-volv-pvc 
Name:          longhorn-volv-pvc
Namespace:     default
StorageClass:  longhorn-v2-data-engine
Status:        Pending
Volume:        
Labels:        <none>
Annotations:   volume.beta.kubernetes.io/storage-provisioner: driver.longhorn.io
               volume.kubernetes.io/storage-provisioner: driver.longhorn.io
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      
Access Modes:  
VolumeMode:    Filesystem
Used By:       volume-test
Events:
  Type     Reason                Age                   From                                                                                      Message
  ----     ------                ----                  ----                                                                                      -------
  Warning  ProvisioningFailed    118s (x3 over 2m26s)  driver.longhorn.io_csi-provisioner-7b95bf4b87-lxkbk_7506b43d-2ae1-49e9-a5fa-f10f5302a938  failed to provision volume with StorageClass "longhorn-v2-data-engine": rpc error: code = Internal desc = Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [message=failed to create volume: unable to create volume pvc-199034b4-c567-4dea-969b-20cefb48a0eb: admission webhook "validator.longhorn.io" denied the request: cannot find engine image by longhornio/longhorn-instance-manager:master-head, code=Internal Server Error, detail=] from [http://longhorn-backend:9500/v1/volumes]
  Normal   Provisioning          22s (x8 over 2m29s)   driver.longhorn.io_csi-provisioner-7b95bf4b87-lxkbk_7506b43d-2ae1-49e9-a5fa-f10f5302a938  External provisioner is provisioning volume for claim "default/longhorn-volv-pvc"
  Warning  ProvisioningFailed    22s (x5 over 2m29s)   driver.longhorn.io_csi-provisioner-7b95bf4b87-lxkbk_7506b43d-2ae1-49e9-a5fa-f10f5302a938  failed to provision volume with StorageClass "longhorn-v2-data-engine": rpc error: code = Internal desc = Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [code=Internal Server Error, detail=, message=failed to create volume: unable to create volume pvc-199034b4-c567-4dea-969b-20cefb48a0eb: admission webhook "validator.longhorn.io" denied the request: cannot find engine image by longhornio/longhorn-instance-manager:master-head] from [http://longhorn-backend:9500/v1/volumes]
  Normal   ExternalProvisioning  9s (x12 over 2m29s)   persistentvolume-controller                                                               waiting for a volume to be created, either by external provisioner "driver.longhorn.io" or manually created by system administrator
```

**Comment by derekbit** [comment]
> But v2 volume can't be created:

It is caused by [enforce-engine-upgrade](https://github.com/longhorn/longhorn/issues/5842) feature. Fixed in https://github.com/longhorn/longhorn-manager/pull/2386/commits/14a841dea4c89deac383c22d6a3d093ce1510cbc.

**Comment by yangchiu** [comment]
> It is caused by [enforce-engine-upgrade](https://github.com/longhorn/longhorn/issues/5842) feature. Fixed in [longhorn/longhorn-manager@14a841d](https://github.com/longhorn/longhorn-manager/commit/14a841dea4c89deac383c22d6a3d093ce1510cbc).

Waiting for this commit being merged and then test again. Thank you!

**Comment by yangchiu** [comment]
> > It is caused by [enforce-engine-upgrade](https://github.com/longhorn/longhorn/issues/5842) feature. Fixed in [longhorn/longhorn-manager@14a841d](https://github.com/longhorn/longhorn-manager/commit/14a841dea4c89deac383c22d6a3d093ce1510cbc).
> 

For better tracking. This is the PR which solves the problem: https://github.com/longhorn/longhorn-manager/pull/2386.

**Comment by yangchiu** [comment]
Verified passed on master-head (longhorn-manager be08850) following [the test steps](https://github.com/longhorn/longhorn/issues/6984#issuecomment-1788339464).
